pkg/infra/usagestats/statscollector/service.go

```diff
@@ -91,14 +97,21 @@ func (s *Service) RegisterProviders(usageStatProviders []registry.ProvidesUsageS
 }
 
 func (s *Service) Run(ctx context.Context) error {
-	s.updateTotalStats(ctx)
-	updateStatsTicker := time.NewTicker(time.Minute * 30)
+	sendInterval := time.Second * time.Duration(s.cfg.MetricsTotalStatsIntervalSeconds)
+	nextSendInterval := time.Duration(rand.Intn(MAX_DELAY-MIN_DELAY)+MIN_DELAY) * time.Second
```

Comment:
my intention was to introduce a random delay at the beginning and then execute all the following ones at the standard interval (eg 30s): since the first one is delayed by a random factor all of the will shift by the same factor so they won't occur exactly at the same time

Reply:
a fixed interval is fine as long as the starting time is random